### PR TITLE
SQL UNION documentation

### PR DESCRIPTION
**Pull request by lwandzura**
added union doc, please review

### CODE REVIEW

File: content/influxdb/cloud-iox/union.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+title: The UNION clause
+list_title: The UNION clause
+description: > 
+    Use the `UNION` clause to combine the results of two or more queries into a single set of results.
+menu:
+  influxdb_cloud_iox:
+    name: The UNION clause
+    parent: Explore data using SQL
+weight: 295
+---
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:11]:
```suggestion
---
title: UNION clause
description: > 
    Use the `UNION` clause to combine the results of two or more queries into a single set of results.
menu:
  influxdb_cloud_iox:
    name: UNION clause
    parent: Explore data using SQL
weight: 295
---
```

File: content/influxdb/cloud-iox/union.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+The `UNION` clause combines the results of two or more `SELECT` statements into a single results set.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:13]:
```suggestion
The `UNION` clause combines the results of two or more `SELECT` statements into a single result set.
```

File: content/influxdb/cloud-iox/union.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Key points to keep in mind when using the `UNION` clause:
+
+ - the number of columns in the column list must be the same
+ - the columns in the column list must be in the same order
+ - the data types in the column list must be the same or compatible
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:19]:
```suggestion
**Note when using the `UNION` clause**:

- The number of columns in each result set must be the same
- Columns must be in the same order
- Columns must be the same or compatible data types
```

Where are these guidelines coming from? Just from testing? In my own testing, columns are unioned/merged by their position in the first select statement. The value of the first column in the second result set is used to populate the first column of the unioned result set, but the name of the column is always from the first set.

File: content/influxdb/cloud-iox/union.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Syntax
+
+```sql
+SELECT a, b, c
+FROM measurement_1
+UNION [ALL]
+SELECT a, b, c
+FROM measurement_2
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:32]:
~~~suggestion
```sql
SELECT
  a,
  b,
  c
FROM
  measurement_1
UNION [ALL]
SELECT
  a,
  b,
  c
FROM
  measurement_2
``` 
~~~

File: content/influxdb/cloud-iox/union.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+SELECT a, b, c
+FROM measurement_1
+UNION [ALL]
+SELECT a, b, c
+FROM measurement_2
+``` 
+
+## Examples
+
+Basic `UNION` query:
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:36]:
```suggestion
### Union results from different measurements
```

File: content/influxdb/cloud-iox/union.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+```sql
+SELECT "pH" as "water_pH", "time", "location"
+FROM "h2o_pH"
+UNION 
+SELECT "location", "time", "randtag" 
+FROM "h2o_quality"
+```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:44]:
~~~suggestion
```sql
SELECT
  "pH" AS "water_pH",
  "time",
  "location"
FROM
  "h2o_pH"
UNION
SELECT
  "index",
  "time",
  "location"
FROM
  "h2o_quality"
```
~~~

File: content/influxdb/cloud-iox/union.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+Results:
+
+| location     | time                     | water_ph |
+| :----------- | :----------------------- | :------- |
+| coyote_creek | 2019-08-21T00:00:00.000Z | 8.0      |
+| coyote_creek | 2019-08-21T00:18:00.000Z | 7.0      |
+| coyote_creek | 2019-08-21T00:42:00.000Z | 8.0      |
+
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:52]:
```suggestion
{{< expand-wrapper >}}
{{% expand "View example results" %}}
| water_pH | time                 | location     |
| -------: | :------------------- | :----------- |
|       60 | 2019-08-31T00:54:00Z | coyote_creek |
|       65 | 2019-08-31T01:36:00Z | coyote_creek |
|        9 | 2019-08-31T02:18:00Z | coyote_creek |
|       57 | 2019-08-31T03:30:00Z | coyote_creek |
|       24 | 2019-08-31T04:54:00Z | coyote_creek |
|       75 | 2019-08-31T05:00:00Z | coyote_creek |
{{% /expand %}}
{{< /expand-wrapper >}}
```

File: content/influxdb/cloud-iox/union.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+The query returns the location, time and pH from the 2 tables.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:53]:
```suggestion
```

File: content/influxdb/cloud-iox/union.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+## Syntax
+
+```sql
+SELECT a, b, c
+FROM measurement_1
+UNION [ALL]
+SELECT a, b, c
+FROM measurement_2
+``` 
+
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:33]: Explain the difference between `UNION` and `UNION ALL`: https://dataschool.com/learn-sql/what-is-the-difference-between-union-and-union-all/#:~:text=The%20main%20difference%20between%20UNION,keeps%20all%20records%2C%20including%20duplicates

File: content/influxdb/cloud-iox/union.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+`UNION ALL` query using an `OVER` clause:
+
+```sql
+SELECT 'bottom' as type, time, water_level FROM
+  (SELECT time, "water_level", row_number() OVER (order by water_level) as rn FROM h2o_feet) where rn <= 3 
+UNION ALL 
+SELECT 'top' as type, time, water_level FROM 
+  (SELECT time, "water_level", row_number() OVER (order by water_level DESC) as rn FROM h2o_feet) where rn <= 3
+  ```
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:63]: I think this query needs a little more explaining. There isn't really anything special being done with `UNION` here, but the subqueries are fairly advanced and might confuse users.

File: content/influxdb/cloud-iox/union.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+| location     | time                     | water_ph |
+| :----------- | :----------------------- | :------- |
+| coyote_creek | 2019-08-21T00:00:00.000Z | 8.0      |
+| coyote_creek | 2019-08-21T00:18:00.000Z | 7.0      |
+| coyote_creek | 2019-08-21T00:42:00.000Z | 8.0      |
+
+The query returns the location, time and pH from the 2 tables.
+
+`UNION ALL` query using an `OVER` clause:
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:55]:
```suggestion
### Return the top and bottom three results in a single result set
```

File: content/influxdb/cloud-iox/union.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+  ```
+Results:
+| time                     | type   | water_level |
+| :----------------------- | :----- | ----------- |
+| 2019-08-28T07:24:00.000Z | top    | 9.964       |
+| 2019-08-28T07:18:00.000Z | top    | 9.957       |
+| 2019-08-28T07:30:00.000Z | top    | 9.954       |
+| 2019-08-28T14:30:00.000Z | bottom | -0.61       |
+| 2019-08-29T15:18:00.000Z | bottom | -0.594      |
+| 2019-08-28T14:36:00.000Z | bottom | -0.591      |
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:72]:
```suggestion
{{< expand-wrapper >}}
{{% expand "View example results" %}}
| time                     | type   | water_level |
| :----------------------- | :----- | ----------- |
| 2019-08-28T07:24:00.000Z | top    | 9.964       |
| 2019-08-28T07:18:00.000Z | top    | 9.957       |
| 2019-08-28T07:30:00.000Z | top    | 9.954       |
| 2019-08-28T14:30:00.000Z | bottom | -0.61       |
| 2019-08-29T15:18:00.000Z | bottom | -0.594      |
| 2019-08-28T14:36:00.000Z | bottom | -0.591      |
{{% /expand %}}
{{< /expand-wrapper >}}
```

File: content/influxdb/cloud-iox/union.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+The query returns the type of `water_level`, top or bottom, using the `OVER` clause to window the data to get the top and bottom 3 results.
```
Comment (sanderson) [inline content/influxdb/cloud-iox/union.md:74]: Move the explanation above the query. I think we probably need to explain more of what's happening in the query.

```suggestion
```

**Review by sanderson — CHANGES_REQUESTED** [review]

File: content/influxdb/cloud-iox/union.md (hunk 10 above)
Comment (lwandzura) [reply content/influxdb/cloud-iox/union.md:70]: I'm not happy with the dataset and am struggling to come up with UNION queries that make sense.  In retrospect, I would have like to use the sensors data for samples.  But I feel like it's too close to Jan 31 to change now.

**Review by sanderson — APPROVED** [review]

**Pull request merged**